presto-hive/src/main/java/io/prestosql/plugin/hive/procedure/RegisterPartitionProcedure.java

```diff
@@ -109,10 +105,6 @@ public void registerPartition(ConnectorSession session, ConnectorAccessControl a
 
     private void doRegisterPartition(ConnectorSession session, ConnectorAccessControl accessControl, String schemaName, String tableName, List<String> partitionColumn, List<String> partitionValues, String location)
     {
-        if (!allowRegisterPartition) {
-            throw new PrestoException(PERMISSION_DENIED, "register_partition procedure is disabled");
```

Comment:
That's a security issue, as register_partition allows one to access arbitrary data location

Reply:
Yes but if the user doesn't have write access to the table, then it will be restricted.

Reply:
> That's a security issue, as register_partition allows one to access arbitrary data location

@Praveen2112 It is not about table that user has access to. It is about accessing locations that user do not have access.

Consider user Bob has full access to tables in schema `bob`. Then he could register a partition in one of these tables to point to location of tables from schema `alice`. So it look like we cannot remove this configuration toggle.